packages/api/internal/cache/instance/monitoring.go

```diff
@@ -8,10 +8,15 @@ import (
 )
 
 func (c *InstanceCache) UpdateCounter(instance InstanceInfo, value int64) {
+	if c.counter == nil {
```

Comment:
If the InstanceCache was created via `NewInstanceCache` it should be always initialized. The same checks could then be for the `c.cache` everywhere.